test/python/circuit/test_parameters.py

```diff
@@ -63,6 +63,16 @@ def test_get_parameters(self):
         self.assertIs(theta, next(iter(vparams)))
         self.assertIs(rxg, vparams[theta][0][0])
 
+    def test_is_parameterized(self):
+        """Test checking if a gate is parameterized"""
+        from qiskit.extensions.standard.h import HGate
+        from qiskit.extensions.standard.rx import RXGate
+        theta = Parameter('θ')
+        rxg = RXGate(theta)
+        self.assertTrue(rxg.is_parameterized())
+        h_gate = HGate()
+        self.assertFalse(h_gate.is_parameterized())
```

Comment:
Could you also add a test for a bound parameter? e.g. `ParameterExpression(3.25)`

Reply:
Ok working on it

Reply:
Added a bound parameter test.

The previous build (triggered by merging master to the branch) failed in Windows, doesn't seem to have anything to do with my code.

Reply:
The syntax for generating and binding a  `ParameterExpression` should look like:
```
>>> theta = Parameter('θ')
>>> theta
Parameter(θ)
>>> theta_sq = theta * theta
>>> theta_sq
ParameterExpression(θ**2)
>>> theta.bind({theta: 3.14})
ParameterExpression(3.14000000000000)
>>> theta_sq.bind({theta: 3.14})
ParameterExpression(9.85960000000000)
```
The constructor for `ParameterExpression` isn't intended to be called directly by users (only through arithmetic operations on other `Parameter`s) though we don't enforce this.

Reply:
Do you want me to change the test to read like that?

Reply:
E.g.,
```
 def test_is_parameterized(self):
        """Test checking if a gate is parameterized (bound/unbound)"""
        from qiskit.extensions.standard.h import HGate
        from qiskit.extensions.standard.rx import RXGate
        theta = Parameter('θ')
        rxg = RXGate(theta)
        self.assertTrue(rxg.is_parameterized())
        theta.bind({theta: 3.14})
        rxg = RXGate(theta)
        self.assertTrue(rxg.is_parameterized())
        h_gate = HGate()
        self.assertFalse(h_gate.is_parameterized())
```

Reply:
That'd be great.

Reply:
changed it / tested / pushed, @kdk

Reply:
Thanks for the quick fix, one more small change (sorry, I didn't get your example code before I commented and this wasn't clear from my example code.) The `bind` operation isn't applied in place, it returns a new `ParameterExpression` with the binding applied, so the tests would look like

```
        theta = Parameter('θ')
        rxg = RXGate(theta)
        self.assertTrue(rxg.is_parameterized())
        theta_bound = theta.bind({theta: 3.14})
        rxg = RXGate(theta_bound)
        self.assertTrue(rxg.is_parameterized())
        h_gate = HGate()
        self.assertFalse(h_gate.is_parameterized())
```

Reply:
ok, testing and will push

Reply:
@kdk pushed corrected test

Reply:
:+1: Looks good, thanks for taking care of it.